File: portfolio-service/src/main/java/com/satheesh/portfolio/service/ChatService.java
package com.satheesh.portfolio.service;

import com.satheesh.portfolio.dto.ChatMessageDTO;

/**
 * @author Satheesh Kumar P
 * @since 2026-07-27
 * @version 1.0.0
 * 
 * @description Service interface for processing Spring AI Gemini Chatbot messages.
 */
public interface ChatService {

    /**
     * Processes a user chat prompt and returns an AI response.
     * 
     * @param chatDTO Input message DTO
     * @param ipAddress Client IP address
     * @return AI response string
     */
    String processChat(ChatMessageDTO chatDTO, String ipAddress);
}

File: frontend/src/services/chatService.js
import api from './api';

/**
 * @author Satheesh Kumar P
 * @since 2026-07-27
 * @version 1.0.0
 * 
 * @description API service module for interacting with Spring AI Gemini Chatbot.
 */
export const chatService = {
  /**
   * Sends user chat prompt to Spring AI backend.
   * @param {string} message User chat prompt
   * @param {string} sessionId Optional conversation session ID
   * @returns {Promise<Object>} { reply: string }
   */
  sendMessage: async (message, sessionId = null) => {
    const response = await api.post('/chat', { message, sessionId });
    return response.data;
  },
};
